File: src/scitrader.marketdataserver/src/main/java/com/scitrader/marketdataserver/common/Model/PriceBar.java
package com.scitrader.marketdataserver.common.Model;

import com.jsoniter.annotation.JsonProperty;
import org.joda.time.DateTime;

import java.math.BigDecimal;

public class PriceBar {

  @JsonProperty("time")
  private DateTime time;
  @JsonProperty("open")
  private BigDecimal open;
  @JsonProperty("high")
  private BigDecimal high;
  @JsonProperty("low")
  private BigDecimal low;
  @JsonProperty("close")
  private BigDecimal close;
  @JsonProperty("volume")
  private BigDecimal volume;


  public DateTime getTime() {
    return time;
  }

  public void setTime(DateTime time) {
    this.time = time;
  }

  public BigDecimal getOpen() {
    return open;
  }

  public void setOpen(BigDecimal open) {
    this.open = open;
  }

  public BigDecimal getHigh() {
    return high;
  }

  public void setHigh(BigDecimal high) {
    this.high = high;
  }

  public BigDecimal getLow() {
    return low;
  }

  public void setLow(BigDecimal low) {
    this.low = low;
  }

  public BigDecimal getClose() {
    return close;
  }

  public void setClose(BigDecimal close) {
    this.close = close;
  }

  public BigDecimal getVolume() {
    return volume;
  }

  public void setVolume(BigDecimal volume) {
    this.volume = volume;
  }
}

File: Kotlin/hello-kotlin-world/src/main/java/com/scitrader/marketdataserver/common/Model/PriceBar.kt
package com.scitrader.marketdataserver.common.Model

import com.jsoniter.annotation.JsonCreator
import com.jsoniter.annotation.JsonProperty
import org.joda.time.DateTime

class PriceBar () {

    var time : DateTime = DateTime(0)
    var open : Double = 0.0
    var high : Double = 0.0
    var low : Double = 0.0
    var close : Double = 0.0
    var volume : Double = 0.0

    @JsonCreator
    constructor(@JsonProperty("time") time: DateTime,
                @JsonProperty("open") open : Double,
                @JsonProperty("high") high : Double,
                @JsonProperty("low") low : Double,
                @JsonProperty("close") close : Double,
                @JsonProperty("volume") volume : Double) : this(){
        this.time = time
        this.open = open
        this.high = high
        this.low = low
        this.close = close
        this.volume = volume
    }
}